setup.py

```diff
@@ -87,7 +87,7 @@ def run(self):
         'protobuf>=4.22,<6.0.0',
         'grpcio>=1.56.2,<1.66.0',
     ],
-    python_requires='>=3.8, <3.12',
+    python_requires='>=3.10, <3.12',
```

Comment:
`<3.13` :)

Reply:
I have reverted back to Python 3.11 from 3.12 because the majority of pipelines are failing due to various TensorFlow version issues. Finding a suitable TensorFlow version and fixing the related issues is redundant effort, as the next couple of PRs will update the version to latest.

Reply:
Can you point me to the failing TF runs? Would like to take a look.

Reply:
have look at the pipelines failing at commit [6da85d3](https://github.com/securefederatedai/openfl/pull/1189/commits/6da85d3ed2d3ff4ae00b4b84e803f527bdda6792) above.

Reply:
Can you please have a look at this also https://github.com/securefederatedai/openfl/actions/runs/12154323404/job/33899656445. It is related to Numpy version

Reply:
> Can you please have a look at this also https://github.com/securefederatedai/openfl/actions/runs/12154323404/job/33899656445. It is related to Numpy version

Is this related to 3.12 failure, or happens in 3.11?